capsules/src/ble_advertising_driver.rs

```diff
@@ -380,11 +267,11 @@ impl AlarmData {
 }
 
 pub struct App {
-    advertisement_buf: Option<kernel::AppSlice<kernel::Shared, u8>>,
     app_write: Option<kernel::AppSlice<kernel::Shared, u8>>,
     app_read: Option<kernel::AppSlice<kernel::Shared, u8>>,
+    address: [u8; PACKET_ADDR_LEN],
+    pdu_type: u8,
```

Comment:
We should have an enum for this!

Reply:
I sort of agree (was getting there), however it's worth discussing whether we actually _should_ enforce which PDU types the user passes in. But will definitely do that for now.

Reply:
Good point, we could probably do some basic checks with help from the `BleState`.

But I think we should move all different `ble_types` into a different file to keep the advertisement driver as short as possible!